iceoryx_examples/icedelivery/iox_subscriber_with_history.cpp

```diff
@@ -43,32 +43,30 @@ int main()
     // publisher has send. The history ensures that we at least get the last 5
     // samples sent by the publisher when we subscribe (if at least 5 were already sent).
     subscriberOptions.historyRequest = 5U;
-    iox::popo::TypedSubscriber<RadarObject> typedSubscriber({"Radar", "FrontLeft", "Object"}, subscriberOptions);
-    typedSubscriber.subscribe();
+    iox::popo::TypedSubscriber<RadarObject> subscriber({"Radar", "FrontLeft", "Object"}, subscriberOptions);
+    subscriber.subscribe();
 
     // run until interrupted by Ctrl-C
     while (!killswitch)
     {
-        if (typedSubscriber.getSubscriptionState() == iox::SubscribeState::SUBSCRIBED)
+        if (subscriber.getSubscriptionState() == iox::SubscribeState::SUBSCRIBED)
         {
             bool hasMoreSamples = true;
             // Since we are checking only every second but the publisher is sending every
             // 400ms a new sample we will receive here more then one sample.
             do
             {
-                typedSubscriber.take()
-                    .and_then([](iox::popo::Sample<const RadarObject>& object) {
-                        std::cout << "Got value: " << object->x << std::endl;
-                    })
-                    .if_empty([&] { hasMoreSamples = false; });
+                subscriber.take_1_0()
+                    .and_then([](auto& object) { std::cout << "Got value: " << object->x << std::endl; })
+                    .or_else([&](auto&) { hasMoreSamples = false; });
```

Comment:
This is semantically different from the old code. But I think more appropriate to prevent an infinite loop it take always returns an error.

Reply:
It is, but if there is an error there are also no samples...any proposal how it should behave?

Reply:
I already think this is a better solution. It also breaks the loop in case of an error.

Reply:
One could distinguish between the error case and no data, but this is already done in the typed subscriber example.